sdk/table/azure-table/azure/table/_table_client.py

```diff
@@ -0,0 +1,575 @@
+                value.expiry = serialize_iso(value.expiry)
+            identifiers.append(SignedIdentifier(id=key, access_policy=value))
+        signed_identifiers = identifiers  # type: ignore
+        try:
+            self._client.table.set_access_policy(
+                table=self.table_name,
+                table_acl=signed_identifiers or None,
+                **kwargs)
+        except HttpResponseError as error:
+            process_storage_error(error)
+
+    @distributed_trace
+    def get_table_properties(
+            self,
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> "models.TableServiceProperties"
+        """Gets the properties of an account's Table service,
+        including properties for Analytics and CORS (Cross-Origin Resource Sharing) rules.
+
+        :keyword callable cls: A custom type or function that will be passed the direct response
+        :return: TableServiceProperties, or the result of cls(response)
+        :rtype: ~azure.table.models.TableServiceProperties
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        timeout = kwargs.pop('timeout', None)
+        request_id_parameter = kwargs.pop('request_id_parameter', None)
+        try:
+            response = self._client.service.get_properties(
+                timeout=timeout,
+                request_id_parameter=request_id_parameter,
+                **kwargs)
+            return response
+        except HttpResponseError as error:
+            process_storage_error(error)
+
+    @distributed_trace
+    def delete_entity(
+            self,
+            partition_key,
+            row_key,
+            etag=None,
+            match_condition=None,
+            **kwargs
+    ):
+        # type: (...) -> None
+        """Deletes the specified entity in a table.
+
+        :param match_condition: MatchCondition
+        :type match_condition: ~azure.core.MatchConditions
+        :param etag: Etag of the entity
+        :type etag: str
+        :param partition_key: The partition key of the entity.
+        :type partition_key: str
+        :param row_key: The row key of the entity.
+        :type row_key: str
+        :keyword callable cls: A custom type or function that will be passed the direct response
+        :return: None, or the result of cls(response)
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+        if_match, if_not_match = _get_match_headers(kwargs=dict(kwargs, etag=etag, match_condition=match_condition),
+                                                    etag_param='etag', match_param='match_condition')
+
+        try:
+            self._client.table.delete_entity(
+                table=self.table_name,
+                partition_key=partition_key,
+                row_key=row_key,
+                if_match=if_match or if_not_match or '*',
+                **kwargs)
+        except ResourceNotFoundError:
```

Comment:
seems like we should just let it raise the ResourceNotFoundError, no need to catch